webapp/lib/controller.dart

```diff
@@ -383,16 +391,24 @@ void command(UIAction action, Data actionData) {
     case UIAction.projectSelected:
       ProjectData projectData = actionData;
       selectedProject = projectData.project;
-      listenForNeedsReplyMetrics(selectedProject);
-      listenForDriverMetrics(selectedProject, DRIVERS);
-      updateNeedsReplyCharts(filterNeedsReplyData(needsReplyDataList));
-      updateSystemEventsCharts(filterSystemEventsData(systemEventsDataMap));
       view.contentView.clearDriverCharts();
       driverMetricsFilters.clear();
       driverYUpperLimitFilters.clear();
-      updateDriverCharts(filterDriversData(driversDataMap));
+      switch (selectedTab) {
+        case ChartType.conversation:
+          listenForNeedsReplyMetrics(selectedProject);
+          updateNeedsReplyCharts(needsReplyDataList);
+          break;
+        case ChartType.driver:
+          listenForDriverMetrics(selectedProject, DRIVERS);
+          updateDriverCharts(driversDataMap);
+          break;
+        case ChartType.system:
+          listenForSystemEvents(PROJECTS);
+          updateSystemEventsCharts(systemEventsDataMap);
+          break;
+      }
```

Comment:
Is this the same as the contents of `_updateChartsView`? if yes, can we just call that instead?

Reply:
There is a comment you left of skipping updating system metrics because they're are project independent, that's why I didn't call `_updateChartsView` cause it updates it.

Reply:
Ah yes, I remember. How about changing  `_updateChartsView` to take an optional `skipUpdateSystemMetricsChart` flag that is by default set to false? Basically I'm trying to reduce the amount of similar but not quite identical code we're generating.

Reply:
Good idea. I've refactored it to do that